platform/linux-generic/odp_pool.c

```diff
@@ -296,7 +282,9 @@ static void init_buffers(pool_t *pool)
 		memset(buf_hdr, 0, (uintptr_t)data - (uintptr_t)buf_hdr);
 
 		/* Initialize buffer metadata */
-		buf_hdr->index = i;
+		buf_hdr->index.u32    = 0;
+		buf_hdr->index.pool   = pool->pool_idx;
+		buf_hdr->index.buffer = i;
```

Comment:
Since you're initializing `index.pool` and `index.buffer` there's no need to set `index.u32` here.

Reply:
It's there to ensure that all bits are zero also when someone would modify the bitfield from two to three fields later on. Similarly to memset() zero is used for struct inits.

Reply:
Plus this is done only once - in pool create phase.